src/components/AvatarWithIndicator.js

```diff
@@ -13,37 +19,132 @@ const propTypes = {
+        this.stopSyncIndicator = this.stopSyncIndicator.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.isSyncing) {
+            this.startSyncIndicator();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.isSyncing && this.props.isSyncing) {
+            this.startSyncIndicator();
+        } else if (prevProps.isSyncing && !this.props.isSyncing) {
+            this.stopSyncIndicator();
+        }
+    }
+
+    componentWillUnmount() {
+        this.stopSyncIndicator();
+    }
+
+    /**
+     * We need to manually loop the animations as `useNativeDriver` does not work well with Animated.loop.
+     *
+     * @memberof AvatarWithIndicator
+     */
+    startRotation() {
+        this.rotate.setValue(0);
+        Animated.timing(this.rotate, {
+            toValue: 1,
+            duration: 2000,
+            easing: Easing.linear,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start(({finished}) => {
+            if (finished) {
+                this.startRotation();
+            }
+        });
+    }
+
+    /**
+     * Start Animation for Indicator
+     *
+     * @memberof AvatarWithIndicator
+     */
+    startSyncIndicator() {
+        this.startRotation();
+        Animated.spring(this.scale, {
+            toValue: 1.666,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start();
+    }
+
+    /**
+     * Stop Animation for Indicator
+     *
+     * @memberof AvatarWithIndicator
+     */
+    stopSyncIndicator() {
+        Animated.spring(this.scale, {
+            toValue: 1,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start(() => {
+            this.rotate.resetAnimation();
+            this.scale.resetAnimation();
+            this.rotate.setValue(0);
+        });
+    }
+
+    render() {
+        const indicatorStyles = [
+            styles.alignItemsCenter,
+            styles.justifyContentCenter,
+            this.props.size === 'large' ? styles.statusIndicatorLarge : styles.statusIndicator,
+            this.props.isActive ? styles.statusIndicatorOnline : styles.statusIndicatorOffline,
+            getSyncingStyles(this.rotate, this.scale),
+        ];
+
+        return (
+            <View
+                style={[this.props.size === 'large' ? styles.avatarLarge : styles.sidebarAvatar]}
+            >
+                <Avatar
+                    style={[this.props.size === 'large' ? styles.avatarLarge : null]}
+                    source={this.props.source}
+                />
+                <Animated.View style={StyleSheet.flatten(indicatorStyles)}>
+                    {this.props.isSyncing && (
+                        <Icon
+                            src={Sync}
+                            fill={themeColors.textReversed}
+                            width={6}
```

Comment:
Just to make sure I am following, when the indicator is in syncing mode, the small white icon that spins is shown at 6x6?

Reply:
I am not setting the width & height for the indicator. Instead, I use scaling which is Smooth. So when we scale the green dot, all its children including the icon will scale. 8px icon was looking big when scaling so I have adjusted it and found 6px is looking fine.

Scaling using fixed px is not smooth as the indicator is absolutely placed thus when we increase width & height, It changes location and does not look good. Oppose to what scaling keeps the center and scale uniformly.

Reply:
Ah, that makes sense to me and sounds good!